### ISSUE TITLE
nix-doom-emacs uses incorrect format-all module?

### ISSUE THREAD

**Issue by Mic92** [vlaci/nix-doom-emacs#285]
It seems that using nix-doom-emacs leads to using an unsupported version of format-all instead of the one pinned by doom-emacs.
This leads to https://github.com/hlissner/doom-emacs/pull/5287

**Comment by vlaci** [comment]
Yeah, a fundamental issue right now that the the derivation doesn't take pinned versions into account. I have started playing around with the idea to use [`builtins.fetchGit`](https://nixos.org/manual/nix/stable/#builtin-fetchGit) to acquire sources. It is no problem to get the repo details but it'd need some investigation.

**Comment by Mic92** [comment]
I am now using an unpackaged doom-nix like this again: https://github.com/Mic92/dotfiles/blob/b02ac9f6bed1ff7bd5ec7bce765d4004a012e8db/nixpkgs-config/modules/emacs/default.nix#L29

This has some advantages:

- nix-doom-emacs import from derivation is kind of slow
- it uses doom's pinned revision, which are better tested
- working native compilation
- faster configuration updates, since change configuration does not involve rebuild doom-emacs
